app/controllers/api/v1/admin/invitations_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Admin
+      class InvitationsController < ApiController
+        before_action only: %i[create] do
+          ensure_authorized('ManageUsers')
+        end
+
+        # POST /api/v1/admin/invitations
+        def create
+          params[:invitations][:emails].split(',').each do |email|
+            invitation = Invitation.find_or_initialize_by(email:, provider: current_provider).tap do |i|
+              i.updated_at = Time.zone.now
```

Comment:
This isn't an issue for this scenario, but we can find or create the record and then `touch` it?
That will probably fire an extra query on creation but it should be light weight,
The advantage is that rails will propaperly manage that and if the model has any associations it will touch it either (this isn't the case probably),